test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -10589,6 +10589,501 @@ BOOST_AUTO_TEST_CASE(snark)
+BOOST_AUTO_TEST_CASE(fixed_variable_exponentiation)
+{
+	char const* sourceCode = R"(
+    contract test {
+		fixed a = -3.5;
+		fixed b = 3;
+
+        function f() public returns (fixed) {
+            return a ** b;
+        }
+    }
+    )";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(createFixed(-42875, 1000, 19)));
+}*/
+
+BOOST_AUTO_TEST_CASE(ufixed_comparison)
+{
+	char const* sourceCode = R"(
+	contract test {
+		ufixed a = 3.5;
+		ufixed b = 1.25;
+
+		function f() public returns (bool) {
+			return a > b;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(true));
+}
+
+BOOST_AUTO_TEST_CASE(fixed_comparison)
+{
+	char const* sourceCode = R"(
+	contract test {
+		fixed a = -3.5;
+		fixed b = -1.25;
+
+		function f() public returns (bool) {
+			return a > b;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(false));
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_conversion)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f(ufixed128x19 x) public returns (ufixed256x30) {
+			return x;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	testContractAgainstCppOnRange("f(ufixed128x19)", [](u256 const &a) -> u256 { return a * pow(u256(10), 30 - 19); }, 0, 100);
+}
+
+BOOST_AUTO_TEST_CASE(fixed_conversion)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f(fixed128x19 x) public returns (fixed256x30) {
+			return x;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	testContractAgainstCppOnRange("f(fixed128x19)", [](u256 const &a) -> u256 { return a * pow(u256(10), 30 - 19); }, -50, -1);
+	testContractAgainstCppOnRange("f(fixed128x19)", [](u256 const &a) -> u256 { return a * pow(u256(10), 30 - 19); }, 0, 50);
+}
+
+BOOST_AUTO_TEST_CASE(uint_ufixed_conversion)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f(uint8 x) public returns (ufixed256x19) {
+			return x;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	testContractAgainstCppOnRange("f(uint8)", [](u256 const &a) -> u256 { return a * pow(u256(10), 19); }, 0, 100);
+}
+
+BOOST_AUTO_TEST_CASE(int_fixed_conversion)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f(int8 x) public returns (fixed256x19) {
+			return x;
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	testContractAgainstCppOnRange("f(int8)", [](u256 const &a) -> u256 { return a * pow(u256(10), 19); }, -50, -1);
+	testContractAgainstCppOnRange("f(int8)", [](u256 const &a) -> u256 { return a * pow(u256(10), 19); }, 0, 50);
+}
+
+BOOST_AUTO_TEST_CASE(ufixed_uint_conversion)
+{
+	char const* sourceCode = R"(
+	contract test {
+		function f(ufixed256x19 x) public returns (uint256) {
+			return uint256(x);
+		}
+	}
+	)";
+	compileAndRun(sourceCode, 0, "test");
+	testContractAgainstCppOnRange("f(ufixed256x19)", [](u256 const &a) -> u256 { return 1 + 0 * a; }, pow(u256(10), 19), pow(u256(10), 19) + 10);
```

Comment:
Could you test the range `10**19 - 10` until `10**10 + 10`? Only that way, we find out whether the implementation rounds or truncates.